core/includes/update.inc

```diff
@@ -294,6 +294,46 @@ function update_module_add_to_system($modules = array()) {
   }
 }
 
+function update_migration_enable_dependencies() {
+  if (backdrop_get_installed_schema_version('system') > 7000 && !state_get('update_migration_dependencies', FALSE)) {
+    
+    // Get list of enabled modules. 
+    $result = db_query("SELECT name FROM {system} WHERE type = :type AND status = :status", array(':type' => 'module', ':status' => 1));
+    
+    // Get list of modules to be enabled.
+    $enabled_modules = array_keys($result->fetchAllAssoc('name'));
+    // Get information about all modules available
+    $files = system_rebuild_module_data();
+
+    $modules_to_enable = array();
+    foreach ($enabled_modules as $enabled_module) {
+      if (!empty($files[$enabled_module]->requires)) {
+        $required_modules = array_keys($files[$enabled_module]->requires);
+        // Add required modules that are not enabled.
+        $modules_to_enable = array_merge($modules_to_enable, array_diff($required_modules, $enabled_modules));
+      }
+    }
+    // Remove duplicates.
+    $modules_to_enable = array_unique($modules_to_enable);
+    if (!empty($modules_to_enable)) {
+      module_enable($modules_to_enable);
+      $modules_string = implode(', ', $modules_to_enable);
+
+      state_set('update_migration_dependencies', TRUE);
+      $requirements['enable_modules_on_migrate'] = array(
+        'title' => t('Modules enabled automatically for migration.'),
+        'description' => t('The followig Backdrop modules needed to be enabled in order for the site to be migrated successully: %modules.', array('%modules' => $modules_string)),
```

Comment:
Lets do something like this here instead (some typos + render the modules as a proper list):

```suggestion
        'description' => t('The following Backdrop modules needed to be enabled in order for the site to be migrated successfully: %modules.', array('%modules' => theme('item_list', array('items' => $modules_string)))),
```

Reply:
...would be even better if we had the human-readable names of the modules instead of machine names.